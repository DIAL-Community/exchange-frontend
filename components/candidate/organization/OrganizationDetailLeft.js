import Bookmark from '../../shared/common/Bookmark'
import Comment from '../../shared/common/Comment'
import Share from '../../shared/common/Share'
import { ObjectType } from '../../utils/constants'
import OrganizationDetailHeader from './fragments/OrganizationDetailHeader'

const OrganizationDetailLeft = ({ scrollRef, organization }) => {
  return (
    <div className='bg-dial-slate-100 lg:h-full'>
      <div className='flex flex-col gap-y-3 px-4 lg:px-6 lg:py-3'>
        <OrganizationDetailHeader organization={organization}/>
        <hr className='border-b border-dial-slate-200'/>
        <div className='hidden lg:block flex flex-col gap-y-3'>
          <Bookmark object={organization} objectType={ObjectType.ORGANIZATION} />
          <hr className='border-b border-dial-slate-200'/>
          <Share />
          <hr className='border-b border-dial-slate-200'/>
          <Comment entityKey={'ui.candidateOrganization.label'} scrollRef={scrollRef} />
          <hr className='border-b border-dial-slate-200'/>
        </div>
      </div>
    </div>
  )
}

export default OrganizationDetailLeft
