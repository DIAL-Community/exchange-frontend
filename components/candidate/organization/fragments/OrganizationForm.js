import React, { useState, useCallback, useContext, useRef } from 'react'
import { useRouter } from 'next/router'
import { useMutation } from '@apollo/client'
import { useIntl } from 'react-intl'
import { FaSpinner } from 'react-icons/fa6'
import { Controller, useForm } from 'react-hook-form'
import ReCAPTCHA from 'react-google-recaptcha'
import { ToastContext } from '../../../../lib/ToastContext'
import { useUser } from '../../../../lib/hooks'
import Input from '../../../shared/form/Input'
import UrlInput from '../../../shared/form/UrlInput'
import Checkbox from '../../../shared/form/Checkbox'
import ValidationError from '../../../shared/form/ValidationError'
import { HtmlEditor } from '../../../shared/form/HtmlEditor'
import { CREATE_CANDIDATE_ORGANIZATION } from '../../../shared/mutation/candidateOrganization'
import { Loading, Unauthorized } from '../../../shared/FetchStatus'
import { DEFAULT_PAGE_SIZE } from '../../../utils/constants'
import {
  CANDIDATE_ORGANIZATION_PAGINATION_ATTRIBUTES_QUERY,
  PAGINATED_CANDIDATE_ORGANIZATIONS_QUERY
} from '../../../shared/query/candidateOrganization'

const OrganizationForm = React.memo(({ organization }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const captchaRef = useRef()
  const [captchaValue, setCaptchaValue] = useState()

  const slug = organization?.slug ?? ''

  const { user, loadingUserSession } = useUser()

  const [mutating, setMutating] = useState(false)
  const [reverting, setReverting] = useState(false)

  const { showSuccessMessage, showFailureMessage } = useContext(ToastContext)

  const router = useRouter()
  const { locale } = router

  const [updateOrganization, { reset }] = useMutation(CREATE_CANDIDATE_ORGANIZATION, {
    refetchQueries: [{
      query: CANDIDATE_ORGANIZATION_PAGINATION_ATTRIBUTES_QUERY,
      variables: { search: '' }
    }, {
      query: PAGINATED_CANDIDATE_ORGANIZATIONS_QUERY,
      variables: { search: '', limit: DEFAULT_PAGE_SIZE, offset: 0 }
    }],
    onCompleted: (data) => {
      const { createCandidateOrganization: response } = data
      if (response.candidateOrganization && response.errors.length === 0) {
        const redirectPath = `/${locale}` +
                             `/candidate/organizations/${response.candidateOrganization.slug}`
        const redirectHandler = () => router.push(redirectPath)
        showSuccessMessage(
          format('toast.submit.success', { entity: format('ui.candidateOrganization.label') }),
          redirectHandler
        )
      } else {
        showFailureMessage(format('toast.submit.failure', { entity: format('ui.candidateOrganization.label') }))
        setMutating(false)
        reset()
      }
    },
    onError: () => {
      showFailureMessage(format('toast.submit.failure', { entity: format('ui.candidateOrganization.label') }))
      setMutating(false)
      reset()
    }
  })

  const [submitter] = organization ? organization.contacts : []

  const {
    handleSubmit,
    register,
    control,
    formState: { errors }
  } = useForm({
    mode: 'onSubmit',
    reValidateMode: 'onChange',
    shouldUnregister: true,
    defaultValues: {
      organizationName: organization?.name ?? '',
      website: organization?.website ?? '',
      description: organization?.description ?? '',
      createStorefront: organization?.createStorefront ?? false,
      name: submitter?.name ?? '',
      email: submitter?.email ?? '',
      title: submitter?.title ?? ''
    }
  })

  const doUpsert = async (data) => {
    if (user) {
      // Set the loading indicator.
      setMutating(true)
      // Pull all needed data from session and form.
      const { userEmail, userToken } = user
      const {
        website,
        description,
        organizationName,
        createStorefront,
        name,
        email,
        title
      } = data
      // Send graph query to the backend. Set the base variables needed to perform update.
      const variables = {
        slug,
        website,
        description,
        organizationName,
        createStorefront,
        name,
        email,
        title,
        captcha: captchaValue
      }

      updateOrganization({
        variables,
        context: {
          headers: {
            'Accept-Language': locale,
            'Authorization': `${userEmail} ${userToken}`
          }
        }
      })
    }
  }

  const updateCaptchaData = (value) => {
    setCaptchaValue(value)
  }

  const cancelForm = () => {
    if (organization) {
      setReverting(true)
      router.push(`/${locale}/candidate/organizations/${slug}`)
    }
  }

  return loadingUserSession
    ? <Loading />
    : user
      ? <form onSubmit={handleSubmit(doUpsert)}>
        <div className='px-4 lg:px-0 py-4 lg:py-6 text-dial-plum'>
          <div className='flex flex-col gap-y-6 text-sm'>
            <div className='text-xl font-semibold'>
              {organization
                ? format('app.editEntity', { entity: organization.name })
                : `${format('app.createNew')} ${format('ui.organization.label')}`}
            </div>
            <div className='flex flex-col gap-y-2'>
              <label className='text-dial-sapphire required-field' htmlFor='organizationName'>
                {format('ui.candidateOrganization.organizationName')}
              </label>
              <Input
                id='organizationName'
                {...register('organizationName', { required: format('validation.required') })}
                placeholder={format('ui.candidateOrganization.organizationName.placeholder')}
                isInvalid={errors.organizationName}
              />
              {errors.organizationName && <ValidationError value={errors.organizationName?.message} />}
            </div>
            <div className='flex flex-col gap-y-2'>
              <label className='text-dial-sapphire required-field' htmlFor='website'>
                {format('organization.website')}
              </label>
              <Controller
                name='website'
                control={control}
                render={({ field: { value, onChange } }) => (
                  <UrlInput
                    id='website'
                    value={value}
                    onChange={onChange}
                    isInvalid={errors.website}
                    placeholder={format('ui.candidateOrganization.website.placeholder')}
                  />
                )}
                rules={{ required: format('validation.required') }}
              />
              {errors.website && <ValidationError value={errors.website?.message} />}
            </div>
            <div className='flex flex-col gap-y-2'>
              <label className='text-dial-sapphire required-field'>
                {format('organization.description')}
              </label>
              <Controller
                name='description'
                control={control}
                render={({ field: { value, onChange } }) => (
                  <HtmlEditor
                    editorId='description-editor'
                    onChange={onChange}
                    initialContent={value}
                    placeholder={format('ui.candidateOrganization.description.placeholder')}
                    isInvalid={errors.description}
                  />
                )}
                rules={{ required: format('validation.required') }}
              />
              {errors.description && <ValidationError value={errors.description?.message} />}
            </div>
            <label className='flex gap-x-2 items-center self-start text-dial-sapphire'>
              <Checkbox {...register('createStorefront')} />
              {format('ui.candidateOrganization.createStorefront')}
            </label>
            <hr className='border-b border-dial-blue-chalk my-3' />
            <div className='flex flex-col gap-y-2'>
              <label className='required-field' htmlFor='name'>
                {format('ui.candidateOrganization.name')}
              </label>
              <Input
                id='name'
                {...register('name', { required: format('validation.required') })}
                placeholder={format('ui.candidateOrganization.name.placeholder')}
                isInvalid={errors.name}
              />
              {errors.name && <ValidationError value={errors.name?.message} />}
            </div>
            <div className='flex flex-col gap-y-2'>
              <label className='required-field' htmlFor='email'>
                {format('ui.candidateOrganization.email')}
              </label>
              <Input
                id='email'
                {...register('email', { required: format('validation.required') })}
                placeholder={format('ui.candidateOrganization.email.placeholder')}
                isInvalid={errors.email}
              />
              {errors.email && <ValidationError value={errors.email?.message} />}
            </div>
            <div className='flex flex-col gap-y-2'>
              <label htmlFor='title'>
                {format('ui.candidateOrganization.title')}
              </label>
              <Input
                id='title'
                {...register('title')}
                placeholder={format('ui.candidateOrganization.title.placeholder')}
              />
            </div>
            <ReCAPTCHA
              sitekey={process.env.NEXT_PUBLIC_CAPTCHA_SITE_KEY}
              onChange={updateCaptchaData}
              ref={captchaRef}
            />
            <div className='flex flex-wrap text-base mt-6 gap-3'>
              <button
                type='submit'
                className='submit-button'
                disabled={mutating || reverting || !captchaValue}
              >
                {`${format('app.submit')} ${format('ui.organization.label')}`}
                {mutating && <FaSpinner className='spinner ml-3' />}
              </button>
              {organization &&
                <button
                  type='button'
                  className='cancel-button'
                  disabled={mutating || reverting}
                  onClick={cancelForm}
                >
                  {format('app.cancel')}
                  {reverting && <FaSpinner className='spinner ml-3' />}
                </button>
              }
            </div>
          </div>
        </div>
      </form>
      : <Unauthorized />
})

OrganizationForm.displayName = 'OrganizationForm'

export default OrganizationForm
