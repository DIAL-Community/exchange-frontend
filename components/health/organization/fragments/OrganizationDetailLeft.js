import Bookmark from '../../shared/Bookmark'
import Comment from '../../shared/Comment'
import Share from '../../shared/Share'
import { ObjectType } from '../../../utils/constants'
import OrganizationDetailHeader from './OrganizationDetailHeader'
import OrganizationDetailNav from './OrganizationDetailNav'
import OrganizationOwner from './OrganizationOwner'

const OrganizationDetailLeft = ({ scrollRef, organization }) => {
  return (
    <div className='bg-white shadow-xl rounded-2xl'>
      <div className='flex flex-col gap-y-3 px-4 lg:px-6 lg:py-3'>
        <OrganizationDetailHeader organization={organization}/>
        <hr className='border-b border-dial-slate-200'/>
        <OrganizationOwner organization={organization}/>
        <hr className='border-b border-dial-slate-200'/>
        <OrganizationDetailNav organization={organization} scrollRef={scrollRef}/>
        <hr className='border-b border-dial-slate-200'/>
        <div className='hidden lg:flex flex-col gap-y-3'>
          <Bookmark object={organization} objectType={ObjectType.ORGANIZATION}/>
          <hr className='border-b border-dial-slate-200'/>
          <Share/>
          <hr className='border-b border-dial-slate-200'/>
          <Comment entityKey={'ui.organization.label'} scrollRef={scrollRef}/>
        </div>
      </div>
    </div>
  )
}

export default OrganizationDetailLeft
