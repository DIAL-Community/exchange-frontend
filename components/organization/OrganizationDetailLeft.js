import Bookmark from '../shared/common/Bookmark'
import Comment from '../shared/common/Comment'
import Share from '../shared/common/Share'
import { ObjectType } from '../utils/constants'
import OrganizationDetailHeader from './fragments/OrganizationDetailHeader'
import OrganizationDetailNav from './fragments/OrganizationDetailNav'
import OrganizationOwner from './fragments/OrganizationOwner'

const OrganizationDetailLeft = ({ scrollRef, organization }) => {
  return (
    <div className='bg-dial-slate-100 lg:h-full'>
      <div className='flex flex-col gap-y-3 px-4 lg:px-6 lg:py-3'>
        <OrganizationDetailHeader organization={organization}/>
        <hr className='border-b border-dial-slate-200'/>
        <OrganizationOwner organization={organization}/>
        <hr className='border-b border-dial-slate-200'/>
        <OrganizationDetailNav organization={organization} scrollRef={scrollRef} />
        <hr className='border-b border-dial-slate-200'/>
        <div className='hidden lg:flex flex-col gap-y-3'>
          <Bookmark object={organization} objectType={ObjectType.ORGANIZATION} />
          <hr className='border-b border-dial-slate-200'/>
          <Share />
          <hr className='border-b border-dial-slate-200'/>
          <Comment entityKey={'ui.organization.label'} scrollRef={scrollRef} />
          <hr className='border-b border-dial-slate-200'/>
        </div>
      </div>
    </div>
  )
}

export default OrganizationDetailLeft
