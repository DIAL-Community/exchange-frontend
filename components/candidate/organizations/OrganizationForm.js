import React, { useContext, useState } from 'react'
import { useIntl } from 'react-intl'
import { useRouter } from 'next/router'
import { useMutation } from '@apollo/client'
import { FaSpinner } from 'react-icons/fa'
import ReCAPTCHA from 'react-google-recaptcha'
import { Controller, useForm } from 'react-hook-form'
import { ToastContext } from '../../../lib/ToastContext'
import { CREATE_CANDIDATE_ORGANIZATION } from '../../../mutations/organization'
import Input from '../../shared/Input'
import ValidationError from '../../shared/ValidationError'
import { HtmlEditor } from '../../shared/HtmlEditor'
import { emailRegex } from '../../shared/emailRegex'
import { Loading, Unauthorized } from '../../shared/FetchStatus'
import { useUser } from '../../../lib/hooks'

const OrganizationForm = () => {
  const { formatMessage } = useIntl()
  const format = (id, values) => formatMessage({ id }, { ...values })

  const router = useRouter()

  const { user, loadingUserSession } = useUser()

  const { showToast } = useContext(ToastContext)

  const [mutating, setMutating] = useState(false)
  const [reverting, setReverting] = useState(false)

  const [createCandidateOrganization] = useMutation(CREATE_CANDIDATE_ORGANIZATION, {
    onError: () => {
      showToast(
        format('candidate-organization.submit.failure'),
        'error',
        'top-center',
        false
      )
    },
    onCompleted: () => {
      showToast(
        format('candidate-organization.submit.success'),
        'success',
        'top-center',
        1000,
        null,
        () => router.push('/organizations')
      )
    }
  })

  const { handleSubmit, register, control, formState: { errors } } = useForm({
    mode: 'onBlur',
    reValidateMode: 'onChange',
    shouldUnregister: true,
    defaultValues: {
      name: '',
      description: '',
      organizationName: '',
      website: '',
      email: '',
      title: '',
      captcha: null
    }
  })

  const doUpsert = async (data) => {
    if (user) {
      setMutating(true)

      const { userEmail, userToken } = user
      const { name, description, organizationName, website, email, title, captcha } = data
      const variables = {
        name,
        website,
        organizationName,
        description,
        email,
        title,
        captcha
      }

      createCandidateOrganization({
        variables,
        context: {
          headers: {
            'Accept-Language': router.locale,
            Authorization: `${userEmail} ${userToken}`
          }
        }
      })
    }
  }

  const cancelForm = () => {
    setReverting(true)
    router.push('/organizations')
  }

  return (
    loadingUserSession ? <Loading /> : user ? (
      <div className='flex flex-col'>
        <div className='py-8 px-8'>
          <div id='content' className='sm:px-0 max-w-full mx-auto'>
            <form onSubmit={handleSubmit(doUpsert)}>
              <div className='bg-edit shadow-md rounded px-8 pt-6 pb-12 mb-4 flex flex-col gap-3'>
                <div className='text-2xl font-bold text-dial-blue pb-4'>
                  {format('candidateOrganization.label')}
                </div>
                <div className='flex flex-col lg:flex-row gap-4'>
                  <div className='w-full lg:w-1/2 flex flex-col gap-y-3'>
                    <div className='form-field-wrapper' data-testid='candidate-organization-organization-name'>
                      <label className='form-field-label required-field' htmlFor='organizationName'>
                        {format('candidateOrganization.organizationName')}
                      </label>
                      <Input
                        {...register('organizationName', { required: format('validation.required') })}
                        id='organizationName'
                        placeholder={format('candidateOrganization.organizationName.placeholder')}
                        isInvalid={errors.organizationName}
                      />
                      {errors.organizationName && <ValidationError value={errors.organizationName?.message} />}
                    </div>
                    <div className='form-field-wrapper' data-testid='candidate-organization-website'>
                      <label className='form-field-label required-field' htmlFor='website'>
                        {format('candidateOrganization.website')}
                      </label>
                      <Input
                        {...register('website', { required: format('validation.required') })}
                        id='website'
                        placeholder={format('candidateOrganization.website.placeholder')}
                        isInvalid={errors.website}
                      />
                      {errors.website && <ValidationError value={errors.website?.message} />}
                    </div>
                    <div className='form-field-wrapper' data-testid='candidate-organization-name'>
                      <label className='form-field-label required-field' htmlFor='name'>
                        {format('candidateOrganization.name')}
                      </label>
                      <Input
                        {...register('name', { required: format('validation.required') })}
                        id='name'
                        placeholder={format('candidateOrganization.name.placeholder')}
                        isInvalid={errors.name}
                      />
                      {errors.name && <ValidationError value={errors.name?.message} />}
                    </div>
                    <div className='form-field-wrapper' data-testid='candidate-organization-email'>
                      <label className='form-field-label required-field' htmlFor='email'>
                        {format('candidateOrganization.email')}
                      </label>
                      <Input
                        type='email'
                        {...register('email',
                          {
                            required: format('validation.required'),
                            pattern: { value: emailRegex , message: format('validation.email') }
                          }
                        )}
                        id='email'
                        placeholder={format('candidateOrganization.email.placeholder')}
                        isInvalid={errors.email}
                      />
                      {errors.email && <ValidationError value={errors.email?.message} />}
                    </div>
                    <div className='form-field-wrapper' data-testid='candidate-organization-title'>
                      <label className='form-field-label required-field' htmlFor='title'>
                        {format('candidateOrganization.title')}
                      </label>
                      <Input
                        {...register('title', { required: format('validation.required') })}
                        id='title'
                        placeholder={format('candidateOrganization.title.placeholder')}
                        isInvalid={errors.title}
                      />
                      {errors.title && <ValidationError value={errors.title?.message} />}
                    </div>
                    <Controller
                      name='captcha'
                      control={control}
                      rules={{ required: format('validation.required') }}
                      render={({ field: { onChange, ref } }) => {
                        return (<ReCAPTCHA sitekey={process.env.NEXT_PUBLIC_CAPTCHA_SITE_KEY} ref={ref} onChange={onChange} />)
                      }}
                    />
                    {errors.captcha && <ValidationError value={errors.captcha?.message} />}
                  </div>
                  <div className='w-full lg:w-2/3' style={{ minHeight: '20rem' }} data-testid='candidate-organization-description'>
                    <label className='block text-xl text-dial-blue flex flex-col gap-y-2'>
                      <p className='required-field'> {format('candidateOrganization.description')}</p>
                      <Controller
                        name='description'
                        control={control}
                        rules={{ required: format('validation.required') }}
                        render={({ field: { value, onChange, onBlur } }) => {
                          return (
                            <HtmlEditor
                              editorId={`${name}-editor`}
                              onBlur={onBlur}
                              onChange={onChange}
                              initialContent={value}
                              isInvalid={errors.description}
                              placeholder={format('candidateOrganization.description.placeholder')}
                            />
                          )
                        }}
                      />
                      {errors.description && <ValidationError value={errors.description?.message} />}
                    </label>
                  </div>
                </div>
                <div className='flex flex-wrap font-semibold text-xl lg:mt-8 gap-3'>
                  <button
                    type='submit'
                    data-testid='submit-button'
                    className='submit-button'
                    disabled={mutating || reverting}
                  >
                    {format('candidateProduct.submit')}
                    {mutating && <FaSpinner className='spinner ml-3 inline' />}
                  </button>
                  <button
                    type='button'
                    className='cancel-button'
                    disabled={mutating || reverting}
                    onClick={cancelForm}
                  >
                    {format('app.cancel')}
                    {reverting && <FaSpinner className='spinner ml-3 inline' />}
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    ) : <Unauthorized />
  )}

export default OrganizationForm
