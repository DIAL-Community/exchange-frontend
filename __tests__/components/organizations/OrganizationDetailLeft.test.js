import { useSession } from 'next-auth/client'
import OrganizationDetailLeft from '../../../components/organizations/OrganizationDetailLeft'
import { mockRouterImplementation, mockSessionImplementation, render } from '../../test-utils'
import CustomMockedProvider from '../../utils/CustomMockedProvider'
import { organization } from './data/OrganizationForm'
import { user } from './data/OrganizationOwner'

// Mock next-router calls.
jest.mock('next/dist/client/router')
// Mock the next-auth's useSession.
jest.mock('next-auth/client')

describe('Unit tests for the OrganizationDetailLeft component.', () => {

  beforeAll(() => {
    mockRouterImplementation()
  })

  test('Should match snapshot - unauthorized user.', () => {
    mockSessionImplementation(false)
    const { container } = render(
      <CustomMockedProvider>
        <OrganizationDetailLeft organization={organization}/>
      </CustomMockedProvider>
    )
    expect(container).toMatchSnapshot()
  })

  test('Should match snapshot - user is organization owner.', () => {
    useSession.mockReturnValue([{ user: { user } }, false])
    const { container } = render(
      <CustomMockedProvider>
        <OrganizationDetailLeft organization={organization}/>
      </CustomMockedProvider>
    )
    expect(container).toMatchSnapshot()
  })

  test('Should match snapshot - user is admin.', () => {
    mockSessionImplementation(true)
    const { container } = render(
      <CustomMockedProvider>
        <OrganizationDetailLeft organization={organization}/>
      </CustomMockedProvider>
    )
    expect(container).toMatchSnapshot()
  })
})
