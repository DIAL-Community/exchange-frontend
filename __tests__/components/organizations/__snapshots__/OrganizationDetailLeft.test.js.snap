// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for the OrganizationDetailLeft component. Should match snapshot - unauthorized user. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="block lg:hidden"
  >
    <div
      class="bg-white pb-3 lg:py-4 whitespace-nowrap text-ellipsis overflow-hidden"
    >
      <a
        class="inline text-dial-blue h5"
        href="/"
      >
        Home
      </a>
      <div
        class="inline h5"
      >
         &gt; 
        <a
          class="text-dial-gray-dark"
          href="/"
        />
      </div>
    </div>
  </div>
  <div
    class="h-20"
  >
    <div
      class="w-full flex gap-3"
    >
      <div>
        <img
          alt="Edit"
          class="inline mr-2"
          height="15px"
          src="/icons/comment.svg"
          width="15px"
        />
        <div
          class="text-dial-blue inline"
        >
          Comments
        </div>
      </div>
    </div>
    <div
      class="h4 font-bold py-4"
    >
      Organization
    </div>
  </div>
  <div
    class="bg-white border-t-2 border-l-2 border-r-2 border-dial-gray lg:mr-6 shadow-lg"
  >
    <div
      class="flex flex-row p-1.5 border-b border-dial-gray text-xs font-semibold text-dial-cyan"
    >
      <img
        alt="Logo for: Principles for Digital Development"
        class="mr-2 h-6"
        src="/icons/digiprins/digiprins.png"
      />
      <div
        class="my-auto"
      >
        ENDORSED ON 2000
      </div>
    </div>
    <div
      class="flex flex-col h-80 p-4"
    >
      <div
        class="text-2xl font-semibold w-4/5 md:w-auto lg:w-64 2xl:w-80 text-dial-purple overflow-hidden"
      >
        Test Organization
      </div>
      <div
        class="m-auto"
      >
        <img
          alt="Logo for: Test Organization"
          class="w-40"
          src="http://localhost:3000/assets/organizations/test_organization.png"
        />
      </div>
    </div>
  </div>
  <div
    class="bg-dial-gray-dark text-xs text-dial-gray-light p-6 lg:mr-6 shadow-lg border-b-2 border-dial-gray"
  >
    If you are the owner of this organization, you can update this organization information.
    <div
      class="flex flex-row gap-3"
    >
      <a
        class="text-dial-yellow block mt-2"
        href="https://docs.osc.dial.community/projects/product-registry/en/latest/org_owner.html"
        rel="noreferrer"
        target="_blank"
      >
        Find more information here
      </a>
      <div
        class="border-l border-dial-gray-light mt-2"
      />
      <button
        class="text-dial-yellow block mt-2 border-b border-transparent hover:border-dial-yellow"
        href="/apply-product-owner"
      >
        Apply as Owner
      </button>
    </div>
  </div>
</div>
`;

exports[`Unit tests for the OrganizationDetailLeft component. Should match snapshot - user is admin. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="block lg:hidden"
  >
    <div
      class="bg-white pb-3 lg:py-4 whitespace-nowrap text-ellipsis overflow-hidden"
    >
      <a
        class="inline text-dial-blue h5"
        href="/"
      >
        Home
      </a>
      <div
        class="inline h5"
      >
         &gt; 
        <a
          class="text-dial-gray-dark"
          href="/"
        />
      </div>
    </div>
  </div>
  <div
    class="h-20"
  >
    <div
      class="w-full flex gap-3"
    >
      <button
        class="delete-button gap-x-1.5"
        data-testid="delete-button"
        type="button"
      >
        <svg
          class="text-sm"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.5 5.5A.5.5 0 0 1 6 6v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm2.5 0a.5.5 0 0 1 .5.5v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm3 .5a.5.5 0 0 0-1 0v6a.5.5 0 0 0 1 0V6z"
          />
          <path
            d="M14.5 3a1 1 0 0 1-1 1H13v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V4h-.5a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1H6a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1h3.5a1 1 0 0 1 1 1v1zM4.118 4 4 4.059V13a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4.059L11.882 4H4.118zM2.5 3V2h11v1h-11z"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="text-sm"
        >
          Delete
        </span>
      </button>
      <a
        class="inline-flex items-center gap-x-1.5 bg-dial-blue px-2 py-1 rounded-md text-white"
        data-testid="edit-link"
        href="/en/organizations/test_organization/edit"
      >
        <img
          alt="Edit"
          height="12px"
          src="/icons/edit.svg"
          width="12px"
        />
        <span
          class="text-sm"
        >
          Edit
        </span>
      </a>
      <div>
        <img
          alt="Edit"
          class="inline mr-2"
          height="15px"
          src="/icons/comment.svg"
          width="15px"
        />
        <div
          class="text-dial-blue inline"
        >
          Comments
        </div>
      </div>
    </div>
    <div
      class="h4 font-bold py-4"
    >
      Organization
    </div>
  </div>
  <div
    class="bg-white border-t-2 border-l-2 border-r-2 border-dial-gray lg:mr-6 shadow-lg"
  >
    <div
      class="flex flex-row p-1.5 border-b border-dial-gray text-xs font-semibold text-dial-cyan"
    >
      <img
        alt="Logo for: Principles for Digital Development"
        class="mr-2 h-6"
        src="/icons/digiprins/digiprins.png"
      />
      <div
        class="my-auto"
      >
        ENDORSED ON 2000
      </div>
    </div>
    <div
      class="flex flex-col h-80 p-4"
    >
      <div
        class="text-2xl font-semibold w-4/5 md:w-auto lg:w-64 2xl:w-80 text-dial-purple overflow-hidden"
      >
        Test Organization
      </div>
      <div
        class="m-auto"
      >
        <img
          alt="Logo for: Test Organization"
          class="w-40"
          src="http://localhost:3000/assets/organizations/test_organization.png"
        />
      </div>
    </div>
  </div>
  <div
    class="bg-dial-gray-dark text-xs text-dial-gray-light p-6 lg:mr-6 shadow-lg border-b-2 border-dial-gray"
  >
    If you are the owner of this organization, you can update this organization information.
    <div
      class="flex flex-row gap-3"
    >
      <a
        class="text-dial-yellow block mt-2"
        href="https://docs.osc.dial.community/projects/product-registry/en/latest/org_owner.html"
        rel="noreferrer"
        target="_blank"
      >
        Find more information here
      </a>
      <div
        class="border-l border-dial-gray-light mt-2"
      />
      <button
        class="text-dial-yellow block mt-2 border-b border-transparent hover:border-dial-yellow"
        href="/apply-product-owner"
      >
        Apply as Owner
      </button>
    </div>
  </div>
</div>
`;

exports[`Unit tests for the OrganizationDetailLeft component. Should match snapshot - user is organization owner. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="block lg:hidden"
  >
    <div
      class="bg-white pb-3 lg:py-4 whitespace-nowrap text-ellipsis overflow-hidden"
    >
      <a
        class="inline text-dial-blue h5"
        href="/"
      >
        Home
      </a>
      <div
        class="inline h5"
      >
         &gt; 
        <a
          class="text-dial-gray-dark"
          href="/"
        />
      </div>
    </div>
  </div>
  <div
    class="h-20"
  >
    <div
      class="w-full flex gap-3"
    >
      <div>
        <img
          alt="Edit"
          class="inline mr-2"
          height="15px"
          src="/icons/comment.svg"
          width="15px"
        />
        <div
          class="text-dial-blue inline"
        >
          Comments
        </div>
      </div>
    </div>
    <div
      class="h4 font-bold py-4"
    >
      Organization
    </div>
  </div>
  <div
    class="bg-white border-t-2 border-l-2 border-r-2 border-dial-gray lg:mr-6 shadow-lg"
  >
    <div
      class="flex flex-row p-1.5 border-b border-dial-gray text-xs font-semibold text-dial-cyan"
    >
      <img
        alt="Logo for: Principles for Digital Development"
        class="mr-2 h-6"
        src="/icons/digiprins/digiprins.png"
      />
      <div
        class="my-auto"
      >
        ENDORSED ON 2000
      </div>
    </div>
    <div
      class="flex flex-col h-80 p-4"
    >
      <div
        class="text-2xl font-semibold w-4/5 md:w-auto lg:w-64 2xl:w-80 text-dial-purple overflow-hidden"
      >
        Test Organization
      </div>
      <div
        class="m-auto"
      >
        <img
          alt="Logo for: Test Organization"
          class="w-40"
          src="http://localhost:3000/assets/organizations/test_organization.png"
        />
      </div>
    </div>
  </div>
  <div
    class="bg-dial-gray-dark text-xs text-dial-gray-light p-6 lg:mr-6 shadow-lg border-b-2 border-dial-gray"
  >
    If you are the owner of this organization, you can update this organization information.
    <div
      class="flex flex-row gap-3"
    >
      <a
        class="text-dial-yellow block mt-2"
        href="https://docs.osc.dial.community/projects/product-registry/en/latest/org_owner.html"
        rel="noreferrer"
        target="_blank"
      >
        Find more information here
      </a>
      <div
        class="border-l border-dial-gray-light mt-2"
      />
      <button
        class="text-dial-yellow block mt-2 border-b border-transparent hover:border-dial-yellow"
        href="/apply-product-owner"
      >
        Apply as Owner
      </button>
    </div>
  </div>
</div>
`;
